### PR TITLE
[ONNX] Fix sequence handling in graph building

### PR DESCRIPTION
**Pull request by titaiwangms**
Previous to this PR, op.Concat is called without required attributes: axis, and val and arg seems wrongly coded.

### CODE REVIEW

File: torch/onnx/_internal/exporter/_building.py [hunk 1]
```diff
@@ -431,17 +431,17 @@ def _process_python_sequences(
                 elif val is None:
                     # Skip None values
                     continue
-                elif isinstance(arg, (ir.Tensor, ir.TensorProtocol)):
+                elif isinstance(val, (ir.Tensor, ir.TensorProtocol)):
```
Comment (titaiwangms) [inline torch/onnx/_internal/exporter/_building.py:434]: @justinchuby Actually I am not sure about this one being a bug or not.

Comment (justinchuby) [reply torch/onnx/_internal/exporter/_building.py:434]: It's a bug. Thanks for fixing!

**Review by justinchuby — APPROVED** [review]
Thanks - would it be possible to create unit tests for this? Could you share more on how you discovered this?

**Comment by titaiwangms** [comment]
I was testing SmolLM_1_7b on benchmarking:

```
python -m bash_bench.bash_bench_huggingface_big --device cuda --dtype float16 --quiet 0 --memory_peak 0 --exporter onnx_dynamo --model SmolLM_1_7b -r 30 -w 10 --verbose 1 --opt_pattern=ir --verbose 10 --dynamic True
```

**Comment by justinchuby** [comment]
nit: I updated the PR description for it to be more descriptive.

**Comment by titaiwangms** [comment]
> Thanks - would it be possible to create unit tests for this? Could you share more on how you discovered this?

I will update a small e2e test in this PR.

**Comment by titaiwangms** [comment]
I would like to merge this first. I will have the test as follow up.

**Pull request closed without merging**